tests/test_pythonflow.py

```diff
@@ -535,3 +536,18 @@ def test_placeholder_with_kwargs():
         b, c = a
 
     assert graph([b, c], {a: [1, 2]}) == (1, 2)
+
+
+def test_thread_compatibility():
+    def work(event):
+        with pf.Graph() as graph:
+            event.wait()
+    event = threading.Event()
+    thread = threading.Thread(target=work, args=(event,))
+    thread.start()
+    try:
+        with pf.Graph() as graph:
+            event.set()
+    except AssertionError as e:
+        event.set()
```

Comment:
Do we need to call `event.set()` here in case the test fails such that the tests don't hang?